src/cdk-experimental/scrolling/virtual-scroll-strategy.ts

```diff
@@ -37,4 +37,7 @@ export interface VirtualScrollStrategy {
 
   /** Called when the offset of the rendered items changed. */
   onRenderedOffsetChanged();
+
+  /** Get the offset for the given index. */
+  getScrollOffsetForIndex(index: number);
```

Comment:
I think we may want to put the whole `scrollToIndex` function as part of the strategy and just have the viewport delegate. While the autosize strategy might not know exactly where the item is, I think we can create an implementation later on that just kind of guesses. It would require the strategy to have full control over what the scroll method does though, not just spitting back an offset.